app/Http/Controllers/api/v1/RoleController.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace App\Http\Controllers\api\v1;
+
+use App\Enums\CustomStatusCodes;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\RoleRequest;
+use App\Http\Resources\Role as RoleResource;
+use App\Role;
+use Exception;
+use Illuminate\Http\Request;
+use Illuminate\Http\Resources\Json\AnonymousResourceCollection;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Auth;
+
+class RoleController extends Controller
+{
+    public function __construct()
+    {
+        $this->authorizeResource(Role::class, 'role');
+        $this->middleware('check.stale:role,\App\Http\Resources\Role,permissions', ['only' => 'update']);
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return AnonymousResourceCollection
+     */
+    public function index(Request $request)
+    {
+        $resource = Role::query();
+
+        if ($request->has('sort_by') && $request->has('sort_direction')) {
```

Comment:
@rhrt04 Please use same query params as @dennis95stumm. I prefer sort_by and sort_direction over sort_by and order_by as it is more clear what each parameter defines